### DISCUSSION TITLE
Can we use amazon-ecr-credential-helper with buildah

### DISCUSSION THREAD

**Discussion by jaskeerat789** [containers/buildah#5071] [Q&A]
How can we use amazon-ecr-credential-helper or docker-credential-ecr-login as credStore for aws-ecr authentication. I don't want to use aws-cli just for ecr auth. In docker we just set `credStore` key as `ecr-login` in `~/.docker/config.json`. Does buildah has similar option to use these cred helpers ? If so, where how can I use it ?

**Comment by ecerulm** [comment]
Well you can't use `credsStore` but you **can use `credHelpers`**:


This shows that `credsStore` does not work with buildah: 
```
mkdir ~/.docker
echo '{"credsStore":"ecr-login"}' > ~/.docker/config.json
buildah push 111111111111.dkr.ecr.eu-north-1.amazonaws.com/testruben20231013:latest # WON'T WORK, credsStore not supported
Error: pushing image "111111111111.dkr.ecr.eu-north-1.amazonaws.com/testruben20231013:latest" to "docker://111111111111.dkr.ecr.eu-north-1.amazonaws.com/testruben20231013:latest": trying to reuse blob sha256:cb4596cc145400fb1f2aa56d41516b39a366ecdee7bf3f9191116444aacd8c90 at destination: checking whether a blob sha256:cb4596cc145400fb1f2aa56d41516b39a366ecdee7bf3f9191116444aacd8c90 exists in 111111111111.dkr.ecr.eu-north-1.amazonaws.com/testruben20231013: authentication required
```

And this shows that if you use `credHelpers` instead (which is better anyway) the it works:
``` 
echo '{"credHelpers": {"111111111111.dkr.ecr.eu-north-1.amazonaws.com":"ecr-login"}}' > ~/.docker/config.json
 buildah push 111111111111.dkr.ecr.eu-north-1.amazonaws.com/testruben20231013:latest # NOW IT WORKS, because credHelpers is supported
Getting image source signatures
Copying blob cb4596cc1454 done   | 
Copying blob ec68ab009408 done   | 
Copying blob e5eed84b0b92 done   | 
Copying blob e76afd493130 done   | 
Copying blob 82c3ec988e2a done   | 
Copying blob e2128a2a4b6e done   | 
Copying config 01a188de05 done   | 
Writing manifest to image destination
```

The configuration file can be `$HOME/.docker/config.json` but also `$HOME.config/containers/auth.json` and others see [config.go](https://github.com/containers/image/blob/6267de4b83d06c31a65fa8918d4064a9ca8e3018/pkg/docker/config/config.go#L37-L42)
